config/src/config/logger_config.rs

```diff
@@ -86,21 +91,96 @@ impl ConfigSanitizer for LoggerConfig {
     }
 }
 
-/// Returns true iff the tokio-console feature is enabled
-fn is_tokio_console_enabled() -> bool {
-    cfg_if! {
-        if #[cfg(feature = "tokio-console")] {
-            true
-        } else {
-            false
+impl ConfigOptimizer for LoggerConfig {
+    fn optimize(
+        node_config: &mut NodeConfig,
+        local_config_yaml: &Value,
+        _node_type: NodeType,
+        _chain_id: ChainId,
+    ) -> Result<bool, Error> {
+        let logger_config = &mut node_config.logger;
+        let local_logger_config_yaml = &local_config_yaml["logger"];
+
+        // Set the tokio console port
+        let mut modified_config = false;
+        if local_logger_config_yaml["tokio_console_port"].is_null() {
+            // If the tokio-console feature is enabled, set the default port.
+            // Otherwise, disable the tokio console port.
+            if is_tokio_console_enabled() {
+                logger_config.tokio_console_port = Some(DEFAULT_TOKIO_CONSOLE_PORT);
+            } else {
+                logger_config.tokio_console_port = None;
```

Comment:
I think destroying someone's configuration is bad. They set the port, but maybe temporarily turned off the function, but if they come back to turn it on later the port they set will be gone.

Reply:
The line above completely prevents this:
```
if local_logger_config_yaml["tokio_console_port"].is_null() {
...
}
```

If a user specifies the port in their config, we won't make any modifications. Otherwise, this logic defines which value to use. Does that make sense?